specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/DiskMigrations.json

```diff
@@ -0,0 +1,415 @@
+        "modelAsString": true
+      }
+    },
+    "MigrationSubTaskStatus": {
+      "description": "Migration child task status.",
+      "type": "string",
+      "enum": [
+        "Undefined",
+        "Running",
+        "Succeeded",
+        "Failed",
+        "Canceled",
+        "Pending",
+        "Skipped"
+      ],
+      "x-ms-enum": {
+        "name": "MigrationSubTaskStatus",
+        "modelAsString": true
+      }
+    },
+    "MigrationSubTaskProperties": {
+      "description": "Disk migration child task properties.",
+      "type": "object",
+      "properties": {
+        "migrationSubtaskStatus": {
+          "description": "The disk migration child task status.",
+          "$ref": "DiskMigrations.json#/definitions/MigrationSubTaskStatus",
+          "readOnly": true
+        },
+        "reason": {
+          "description": "The reason of task failure.",
+          "type": "string",
+          "readOnly": true
+        },
+        "startTime": {
+          "description": "The task start time.",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "endTime": {
+          "description": "The task end time.",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "targetShare": {
+          "description": "The target share of migration task.",
+          "type": "string",
+          "readOnly": true
+        },
+        "sourceShare": {
+          "description": "The source share of migration task.",
+          "type": "string",
+          "readOnly": true
+        },
+        "targetDiskStateForMigration": {
+          "description": "The disk status.",
+          "$ref": "Disks.json#/definitions/DiskState",
+          "readOnly": true
+        },
+        "diskId": {
+          "description": "The id of disk.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "MigrationSubTaskList": {
+      "description": "List of disk migration tasks.",
+      "type": "array",
+      "items": {
+        "$ref": "DiskMigrations.json#/definitions/MigrationSubTask"
+      }
+    },
+    "MigrationSubTask": {
+      "description": "Disk migration child task.",
+      "type": "object",
+      "properties": {
+        "migrationSubTaskId": {
+          "description": "The id of migration child task.",
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "DiskMigrations.json#/definitions/MigrationSubTaskProperties",
+          "description": "Disk migration child task properties."
+        }
+      }
+    },
+    "DiskMigrationProperties": {
+      "description": "Disk migration job properties.",
+      "type": "object",
+      "properties": {
+        "migrationId": {
+          "description": "The disk migration id.",
+          "type": "string",
+          "readOnly": false
```

Comment:
Since you say that users can modify this property what does it mean to change the migration id?

Reply:
Yes, user can create a DiskMigration object and set the migrationId. and then pass it to the command as input. 

Readonly doesn't only mean "user can change it". It also means it will be serialized back to server. Otherwise, the generated SDK code will not serialize this property.

Reply:
It sounds like the server has a bug.  Properties which should not be changed should be marked readOnly in the spec.

Reply:
Updated.